### PR TITLE
Update Centipede

### PR DESCRIPTION
**Pull request by DonggeLiu**
Update `Centipede` to [`8aaf12`](https://github.com/google/fuzztest/commit/8aaf126c312e414c15e362bd2bd7f1325ebffdbe), which updates corpus distillation features.

### CODE REVIEW

File: infra/base-images/base-builder/compile_centipede [hunk 1]
```diff
@@ -24,7 +24,7 @@ fi
 cp "$CENTIPEDE_BIN_DIR/libcentipede_runner.pic.a" "$LIB_FUZZING_ENGINE"
 
 export CENTIPEDE_FLAGS=`cat "$SRC/fuzztest/centipede/clang-flags.txt" | tr '\n' ' '`
-export LIBRARIES_FLAGS="-Wno-unused-command-line-argument -Wl,-ldl -Wl,-lrt -Wl,-lpthread -Wl,$SRC/fuzztest/centipede/weak.o"
+export LIBRARIES_FLAGS="-Wno-unused-command-line-argument -Wl,-lstdc++ -Wl,-ldl -Wl,-lrt -Wl,-lpthread -Wl,$SRC/fuzztest/centipede/weak.o"
```
Comment (Alan32Liu) [inline infra/base-images/base-builder/compile_centipede:27]: @jonathanmetzman Would it be a bad idea to add `stdlibc++` here?
I recall we do not use `stdlibc++` and do not expect projects to use it, but it is required to fix `skcms` build failure.

**Comment by Alan32Liu** [comment]
Weird, trial-build failed due to
```log
Step #1: CommandException: No URLs matched: gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/*
Step #1: CommandException: 1 file/object could not be transferred.
```

Yet I am pretty sure they are not empty:
```bash
$ gsutil ls gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/
gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/iccprofile_atf.covreport
gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/iccprofile_info.covreport
gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/iccprofile_transform.covreport
```

**Comment by Alan32Liu** [comment]
I presume [this URL](https://github.com/google/oss-fuzz/blob/4c40a91ed0fab9df9da5434b87e776c5645cd9d3/infra/build/functions/build_lib.py#L273) should be quoted?

```bash
# Without quotation:
gsutil -m cp -r gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/* .
no matches found: gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/*

# With quotation:
gsutil -m cp -r 'gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/*' .
Copying gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/iccprofile_atf.covreport...
Copying gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/iccprofile_info.covreport...
Copying gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/iccprofile_transform.covreport...
- [3/3 files][438.8 KiB/438.8 KiB] 100% Done
Operation completed over 3 objects/438.8 KiB.
```

**Comment by Alan32Liu** [comment]
> Weird, trial-build failed due to
> 
> ```
> Step #1: CommandException: No URLs matched: gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/*
> Step #1: CommandException: 1 file/object could not be transferred.
> ```
> 
> Yet I am pretty sure they are not empty:
> 
> ```shell
> $ gsutil ls gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/
> gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/iccprofile_atf.covreport
> gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/iccprofile_info.covreport
> gs://oss-fuzz-coverage/skcms/textcov_reports/20231028/iccprofile_transform.covreport
> ```

Another cause of this is missing the latest report. For example, we query today's report when it is unready.
A potential fix is setting the date to yesterday by changing [this line](https://github.com/google/oss-fuzz/blob/master/infra/build/functions/build_and_run_coverage.py#L293).

**Comment by Alan32Liu** [comment]
Trial-build failures:
* [ ] file: https://oss-fuzz-gcb-logs.storage.googleapis.com/log-fd88f4d4-c43a-4f0a-aff1-63ca9d06657a.txt
* [ ] libultrahdr: https://oss-fuzz-gcb-logs.storage.googleapis.com/log-04c8a353-1f61-4f9d-9408-e2e79bf03c54.txt
* [ ] lua: https://oss-fuzz-gcb-logs.storage.googleapis.com/log-ea74c1ee-bc1f-4d1d-a6ec-23da3dc0f4cd.txt
* [ ] arrow: https://oss-fuzz-gcb-logs.storage.googleapis.com/log-14ece047-27d9-4951-818f-e1796d5b0305.txt
* [ ] hermes: https://oss-fuzz-gcb-logs.storage.googleapis.com/log-f22e5d05-a977-4ee3-92ad-2797a34040bd.txt

**Review by oliverchang — APPROVED** [review]
LGTM but please disable centipede for the breaking project.

**Comment by Alan32Liu** [comment]
Will re-run trial-build in the evening: Today's cov-report is unready yet.

**Comment by jonathanmetzman** [comment]
> Will re-run trial-build in the evening: Today's cov-report is unready yet.

SHoot why are we even doing coverage builds for centipede?

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py all --fuzzing-engines centipede --sanitizers address

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py all --fuzzing-engines centipede --sanitizers address none

File: infra/base-images/base-builder/compile_centipede (hunk 1 above)
Comment (jonathanmetzman) [reply infra/base-images/base-builder/compile_centipede:27]: I think we definitely should not add -lstdc++. We already link against libc++
Let me check on the skcms problem.

**Review by jonathanmetzman — APPROVED** [review]
lgtm

File: infra/base-images/base-builder/compile_centipede (hunk 1 above)
Comment (Alan32Liu) [reply infra/base-images/base-builder/compile_centipede:27]: Thanks!

**Comment by Alan32Liu** [comment]
/gcbrun trial_build.py all --fuzzing-engines centipede --sanitizers address

**Comment by Alan32Liu** [comment]
Testing `Centipede` without `stdlibc++`.
This is likely to fail on CI due to compatibility issues with `skcms` (discussed above).

Testing if this works with other projects via trial build.

**Comment by Alan32Liu** [comment]
Almost all project failed to build due to the same error as `skcms`.
Will add `stdlibc++` back.

Unfortunately, corpus minimization is not supported without this new version of `Centipede`.

**Comment by Alan32Liu** [comment]
/gcbrun trial_build.py all --fuzzing-engines centipede --sanitizers address

**Comment by Alan32Liu** [comment]
/gcbrun trial_build.py all --fuzzing-engines centipede --sanitizers address none

**Comment by DavidKorczynski** [comment]
Am seeing errors building the base builder after this was merged:

```sh
 => [14/30] RUN cd /src &&     git clone https://github.com/google/fuzztest &&     cd fuzztest &&     git checkout 8aaf126c312e414c15e362bd2bd7f1325ebffdbe   1.4s 

 => [15/30] COPY precompile_centipede /usr/local/bin/                                                                                                         0.0s  => ERROR [16/30] RUN precompile_centipede                                                                                                                   70.2s 

------                                                                                                                                                             

 > [16/30] RUN precompile_centipede:                                             

0.307 Precompiling centipedeHit:1 http://archive.ubuntu.com/ubuntu focal InRelease                                                                                 

0.623 Hit:2 http://security.ubuntu.com/ubuntu focal-security InRelease                                                                                             0.706 Hit:3 http://archive.ubuntu.com/ubuntu focal-updates InRelease                                                                                               

0.810 Hit:4 http://archive.ubuntu.com/ubuntu focal-backports InRelease                                                                                             

0.896 Reading package lists...                                                                                                                                     

1.845 Reading package lists...                                                                                                                                     2.724 Building dependency tree...                                                

2.900 Reading state information...                                                                                                                                 

3.074 Suggested packages:                                                        

3.074   libssl-doc                                                                                                                                                 

3.096 The following NEW packages will be installed:                                                                                                                3.097   libssl-dev                                                                                                                                                 

...
OTOBUF_NAMESPACE_CLOSE'

69.68    51 | PROTOBUF_NAMESPACE_CLOSE

69.68       | ^

69.68 bazel-out/k8-opt/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/any_proto/google/protobuf/any.pb.h:52:24: error: expected ';' after top level declarator

69.68    52 | PROTOBUF_NAMESPACE_OPEN

69.68       |                        ^

69.68       |                        ;

69.68 bazel-out/k8-opt/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/any_proto/google/protobuf/any.pb.h:54:1: error: unknown type name 'PROTOBUF_NAMESPACE_CLOSE'

69.68    54 | PROTOBUF_NAMESPACE_CLOSE

69.68       | ^

69.68 bazel-out/k8-opt/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/any_proto/google/protobuf/any.pb.h:55:24: error: expected ';' after top level declarator

69.68    55 | PROTOBUF_NAMESPACE_OPEN

69.68       |                        ^

69.68       |                        ;

69.68 bazel-out/k8-opt/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/any_proto/google/protobuf/any.pb.h:60:14: error: no member named 'PROTOBUF_NAMESPACE_ID' in the global namespace

69.68    60 |     public ::PROTOBUF_NAMESPACE_ID::Message /* @@protoc_insertion_point(class_definition:google.protobuf.Any) */ {

69.68       |            ~~^

69.68 bazel-out/k8-opt/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/any_proto/google/protobuf/any.pb.h:60:37: error: expected class name

69.68    60 |     public ::PROTOBUF_NAMESPACE_ID::Message /* @@protoc_insertion_point(class_definition:google.protobuf.Any) */ {

69.68       |                                     ^

69.68 bazel-out/k8-opt/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/any_proto/google/protobuf/any.pb.h:63:10: error: only virtual member functions can be marked 'override'

69.68    63 |   ~Any() override;

69.68       |          ^~~~~~~~

69.68 bazel-out/k8-opt/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/any_proto/google/protobuf/any.pb.h:64:37: error: no member named 'PROTOBUF_NAMESPACE_ID' in the global namespace

69.68    64 |   explicit PROTOBUF_CONSTEXPR Any(::PROTOBUF_NAMESPACE_ID::internal::ConstantInitialized);

69.68       |                                   ~~^

69.68 fatal error: too many errors emitted, stopping now [-ferror-limit=]

69.68 20 errors generated.

69.83 INFO: Elapsed time: 59.407s, Critical Path: 12.45s

69.83 INFO: 2691 processes: 1217 internal, 2 local, 1472 processwrapper-sandbox.

69.83 ERROR: Build did NOT complete successfully

------

Dockerfile:142

--------------------

 140 |     ENV CENTIPEDE_BIN_DIR=$SRC/fuzztest/bazel-bin

 141 |     COPY precompile_centipede /usr/local/bin/

 142 | >>> RUN precompile_centipede

 143 |     


```
 
There has been a fair amount of changes in the base builder since the last trial run so I assume it's quite likely the old run is no longer valid

**Comment by DavidKorczynski** [comment]
Can confirm reverting this fixes the issue and makes the base builder build again. Put up a diff to revert here: https://github.com/google/oss-fuzz/pull/12160

**Pull request merged**